sympy/discrete/transforms.py

```diff
@@ -48,9 +48,7 @@ def _fourier_transform(seq, symbolic, dps, inverse=False):
 
     ang = -2*pi/n if inverse else 2*pi/n
 
-    if not symbolic:
-        if dps is None:
-            dps = 15
+    if dps is not None:
         ang = ang.evalf(dps + 10)
```

Comment:
`dps + 2` would probably suffice. 10 extra decimals feel like overkill.

Reply:
Okay. As a follow up to efficiency, currently we have `n` calls to `sin`/`cos`.
It can be reduced to `log(n)` calls, by using `w[i] = w[i&(i - 1)] * w[i&-i]` (given all powers of 2 are initialized in `w`), at the cost of error accumulation (which can be accounted for by increasing the precision).

Reply:
The above method has `O(log(n))` error accumulation and `O(n + c*log(n))` complexity (tradeoff).
The current method has error accumulation as `O(1)`, complexity is `O(n * c)`, c is the constant for calculating `sin`/`cos`.
The problem with `w[i]=w[i-1] * w[1]` is that error accumulation is `O(n)` significant, complexity is `O(n)`.